database/vertica/vertica.go

```diff
@@ -0,0 +1,336 @@
+	nurl "net/url"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/hashicorp/go-multierror"
+	_ "github.com/vertica/vertica-sql-go"
+)
+
+func init() {
+	database.Register("vertica", &Vertica{})
+}
+
+var DefaultMigrationsTable = "schema_migrations"
+
+var (
+	ErrDatabaseDirty    = fmt.Errorf("database is dirty")
+	ErrNilConfig        = fmt.Errorf("no config")
+	ErrNoDatabaseName   = fmt.Errorf("no database name")
+	ErrAppendPEM        = fmt.Errorf("failed to append PEM")
+	ErrTLSCertKeyConfig = fmt.Errorf("To use TLS client authentication, both x-tls-cert and x-tls-key must not be empty")
+)
+
+type Config struct {
+	MigrationsTable string
+	DatabaseName    string
+	Schema          string
+}
+
+type Vertica struct {
+	conn   *sql.Conn
+	db     *sql.DB
+	config *Config
+}
+
+func (v *Vertica) Open(url string) (database.Driver, error) {
+	purl, err := nurl.Parse(url)
+	db, err := sql.Open("vertica", migrate.FilterCustomQuery(purl).String())
+	if err != nil {
+		return nil, err
+	}
+
+	migrationsTable := purl.Query().Get("x-migrations-table")
+	if len(migrationsTable) == 0 {
+		migrationsTable = DefaultMigrationsTable
+	}
+	schema := purl.Query().Get("x-schema")
+
+	vx, err := WithInstance(db, &Config{
+		Schema:          schema,
+		DatabaseName:    parseDbName(url),
+		MigrationsTable: migrationsTable,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return vx, nil
+}
+
+func (v *Vertica) Close() error {
+	connErr := v.conn.Close()
+	dbErr := v.db.Close()
+	if connErr != nil || dbErr != nil {
+		return fmt.Errorf("conn: %v, db: %v", connErr, dbErr)
+	}
+	return nil
+}
+
+func (v *Vertica) dropTables() error {
+	// select all tables
+	whereStmt := ""
+	if len(v.config.Schema) > 0 {
+		whereStmt = ` WHERE table_schema='` + v.config.Schema + `'`
+	}
+	query := `SELECT DISTINCT table_name FROM tables` + whereStmt
+	tables, err := v.conn.QueryContext(context.Background(), query)
+	if err != nil {
+		return &database.Error{OrigErr: err, Query: []byte(query)}
+	}
+	defer func() {
+		if errClose := tables.Close(); errClose != nil {
+			err = multierror.Append(err, errClose)
+		}
+	}()
+
+	// delete one table after another
+	tableNames := make([]string, 0)
+	for tables.Next() {
+		var tableName string
+		if err := tables.Scan(&tableName); err != nil {
+			return err
+		}
+		if len(tableName) > 0 {
+			tableNames = append(tableNames, tableName)
+		}
+	}
+
+	if len(tableNames) > 0 {
+
+		// delete one by one ...
+		for _, t := range tableNames {
+			query = "DROP TABLE " + t + " CASCADE"
+			if _, err := v.conn.ExecContext(context.Background(), query); err != nil {
+				return &database.Error{OrigErr: err, Query: []byte(query)}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (v *Vertica) dropViews() error {
+	// select all tables
+	whereStmt := ""
+	if len(v.config.Schema) > 0 {
+		whereStmt = ` WHERE table_schema='` + v.config.Schema + `'`
+	}
+	query := `SELECT DISTINCT table_name FROM views` + whereStmt
```

Comment:
Same thing w/ `v_catalog`

Reply:
I switch both to v_catalog to avoid confusion